packages/aws-cdk-lib/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -149,6 +149,19 @@ export interface QueueProcessingServiceBaseProps {
    */
   readonly scalingSteps?: ScalingInterval[];
 
+  /**
+   * Grace period after scaling activity in seconds.
+   *
+   * Subsequent scale outs during the cooldown period are squashed so that only
+   * the biggest scale out happens.
+   *
+   * Subsequent scale ins during the cooldown period are ignored.
+   *
+   * @see https://docs.aws.amazon.com/autoscaling/application/APIReference/API_StepScalingPolicyConfiguration.html
+   * @default No cooldown
+   */
+  readonly cooldown?: Duration;
```

Comment:
Are we allowed to supply any duration here? Or is there a min/max? I suggest you do some research to figure that out, and then, if there is a limitation of some sort, we should have a synth-time check for that. And then of course also a unit test.

Reply:
Couldn't find a hard max in the docs ([here](https://docs.aws.amazon.com/autoscaling/application/APIReference/API_StepScalingPolicyConfiguration.html#API_StepScalingPolicyConfiguration_Contents) and [here](https://docs.aws.amazon.com/autoscaling/application/userguide/step-scaling-policy-overview.html#step-scaling-cooldown)). 

I then tried to make my own scaling policy in the console and see if there's a max for the cooldown. I didn't get an explicit error but there was weird behavior when I went above 999999999 (the policy would have a different number than the one I set). Should that be our max then? 

As for min as long as its above 0 it should work

Reply:
I think it would be reasonable for 999999999 to be the max if there wasn't any other specified value. At the very least it would prevent unexpected behaviour.

Reply:
Cool, added an error check for the max value